sdk/python/kfp/components/tasks_group.py

```diff
@@ -171,18 +171,23 @@ class ParallelFor(TasksGroup):
 
     Args:
         items: The items to loop over. It can be either a constant Python list or a list output from an upstream task.
-        name: The name of the for loop group.
-        parallelism: The maximum number of concurrent subtasks that can be scheduled for execution.
+        name: Optional; the name of the for loop group.
+        parallelism: Optional; the maximum number of concurrent sub-iterations that can be scheduled for execution. A value of 0 represents unconstrained parallelism (default is unconstrained).
```

Comment:
very small nit:
does `sub-iterations` mean something different than `iterations`

Reply:
not really :') I was struggling to find an appropriate replacement for "subtasks" because I think that could be a little confusing to mean either [the subtasks spawned by the ParallelFor] or [the subtasks defined in the ParallelFor construct itself, e.g. task1].

But I think "iterations" would work nicely in this context.

Reply:
Thanks for explaining and agreed, the difference in meaning is very subtle here.